src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif

```diff
@@ -42,12 +44,15 @@
                 },
                 {
                   "step": 1,
+                  "location": {},
```

Comment:
There is a code change needed in the SDVConverter, looking for it now.

Reply:
I don't think it's a code change. I think if you remove it from the `required` list in the schema, the serializer will refrain from emitting the default value (empty object).

Reply:
So basically there is a combination of property values that lead to none of the Location props having values. Do you want me to try to check for this to eliminate the "{}"?

Reply:
As we discussed: Each time you want to assign to a property of the location object, first check if it already exists, create it if it doesn't, and populate the property.
The result will be that sometimes the value of `codeFlowLocation.location` will be `null`. To prevent the converter from emitting`"location": null`, remove `location` from the list of  `required` properties.